src/main/java/uk/jamierocks/meta/impl/friend/FriendValue.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * This file is part of MetaAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2015, Jamie Mansfield <https://github.com/jamierocks>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package uk.jamierocks.meta.impl.friend;
+
+import uk.jamierocks.meta.api.value.Value;
+
+public abstract class FriendValue<V> extends Friend implements Value<V> {
+
+    public V get(Friend F) throws IllegalArgumentException {
+        if(isFriend(F)) {
+            this.get();
+        }
+        else{
+            throw new IllegalArgumentException("not a friend of "+this.getClass().getName());
+        }
+        return null;
+    }
+
+}
```

Comment:
From what I can tell having a FriendValue is pointless.
If it works like I think it does it should be a detail handled by the valueprocessors.